app/js/tabs.js

```diff
@@ -317,33 +325,68 @@ const addTab = options => {
     } else {
         tabs.appendChild(tab)
     }
-    const webview = document.createElement("webview")
-    webview.setAttribute("link-id", linkId)
     tab.setAttribute("link-id", linkId)
-    if (SETTINGS.get("spell")) {
-        webview.setAttribute("webpreferences", "spellcheck=yes")
-    }
     const color = SETTINGS.get("containercolors").split(",").find(
         c => sessionName.match(c.split("~")[0]))
     if (color) {
         tab.style.color = color.split("~")[1]
     }
-    SESSIONS.create(sessionName)
-    webview.setAttribute("partition", `persist:${sessionName}`)
-    webview.setAttribute("container", sessionName)
-    pages.appendChild(webview)
+
+    const page = document.createElement("div")
+    page.classList.add("webview")
+    page.setAttribute("link-id", linkId)
+    if (options.url) {
+        page.src = UTIL.stringToUrl(options.url)
+    }
+    page.callback = options.callback
+    page.setAttribute("container", sessionName)
+    if (isDevtoolsTab) {
+        page.setAttribute("devtools-for-id", currentPageId)
+    }
+    page.setAttribute("muted", options.muted)
+    pages.appendChild(page)
+    if (options.lazy) {
+        const url = UTIL.stringToUrl(options.url)
+        tab.setAttribute("suspended", "suspended")
+        title.textContent = HISTORY.titleForPage(url) || url
+        favicon.src = FAVICONS.forSite(url) || favicon.src
+    } else {
+        unsuspendPage(page)
+    }
     if (options.switchTo) {
         switchToTab(listTabs().indexOf(tab))
     }
+}
+
+const unsuspendPage = page => {
+    if (page.tagName === "WEBVIEW") {
+        return
+    }
+    tabOrPageMatching(page).removeAttribute("suspended")
+    const webview = document.createElement("webview")
+    ;["link-id", "container", "class", "id", "style"].forEach(
+        attr => webview.setAttribute(attr, page.getAttribute(attr)))
+    if (SETTINGS.get("spell")) {
+        webview.setAttribute("webpreferences", "spellcheck=yes")
+    }
+    const sessionName = page.getAttribute("container")
+    SESSIONS.create(sessionName)
+    webview.setAttribute("partition", `persist:${sessionName}`)
+    const currentPageId = page.getAttribute("devtools-for-id")
+    const isDevtoolsTab = !!currentPageId
     if (isDevtoolsTab) {
         webview.src = "about:blank"
     } else {
         webview.src = UTIL.specialPagePath("newtab")
     }
-    linkId += 1
+    const muted = page.getAttribute("muted") === "true"
+    const url = page.src
+    const callback = page.callback
     webview.addEventListener("dom-ready", () => {
         if (!webview.getAttribute("dom-ready")) {
-            if (options.muted) {
+            const tab = tabOrPageMatching(webview)
+            const title = tab.querySelector("span")
+            if (muted) {
```

Comment:
This if statement can then be changed to check for the attribute, and doesn't need to set it on the tab anymore.